### PR TITLE
Added a descrption attribute to the enumeration values

### PR DESCRIPTION
**Pull request by screig**
I found that the code generation code shed a lot of useful information

### CODE REVIEW

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs [hunk 1]
```diff
@@ -51,7 +51,18 @@ public IEnumerable<EnumerationItemModel> Enums
...
+             
                 var entries = new List<EnumerationItemModel>();
+                
+                Dictionary<string, object> Descriptions = new Dictionary<string, object>();
+                Dictionary<string, object> Titles = new Dictionary<string, object>();
+
+                if (_schema.ExtensionData.ContainsKey("elaboration"))
+                    Descriptions = (Dictionary<string, object>)_schema.ExtensionData["elaboration"];
+
+                if (_schema.ExtensionData.ContainsKey("options"))
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: Why "options"? Is this in the JSON Schema standard?

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs [hunk 2]
```diff
@@ -51,7 +51,18 @@ public IEnumerable<EnumerationItemModel> Enums
...
             get
             {
+             
                 var entries = new List<EnumerationItemModel>();
+                
+                Dictionary<string, object> Descriptions = new Dictionary<string, object>();
+                Dictionary<string, object> Titles = new Dictionary<string, object>();
+
+                if (_schema.ExtensionData.ContainsKey("elaboration"))
+                    Descriptions = (Dictionary<string, object>)_schema.ExtensionData["elaboration"];
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: Why "elaboration"? Is this in the JSON Schema standard?

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs [hunk 3]
```diff
@@ -51,7 +51,18 @@ public IEnumerable<EnumerationItemModel> Enums
         {
             get
             {
+             
                 var entries = new List<EnumerationItemModel>();
+                
+                Dictionary<string, object> Descriptions = new Dictionary<string, object>();
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: Use var

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs [hunk 4]
```diff
@@ -61,11 +72,16 @@ public IEnumerable<EnumerationItemModel> Enums
                             _schema.EnumerationNames.ElementAt(i) :
                             _schema.Type.HasFlag(JsonObjectType.Integer) ? "_" + value : value.ToString();
 
+                        string MyDesc = value.ToString();
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: Use var, rename to "description"

Comment (RSuter) [reply src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: Only set description when it's actually available, no value fallback, then only render attribute if description is available (see below)

File: src/NJsonSchema.CodeGeneration.CSharp/Templates/Enum.liquid [hunk 5]
```diff
@@ -5,10 +5,11 @@
 {{ TypeAccessModifier }} enum {{ Name }}
 {
 {% for enum in Enums -%}
+    [System.ComponentModel.Description("{{ enum.Description }}")]
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration.CSharp/Templates/Enum.liquid]: Only render attribute if description is available...

File: src/NJsonSchema.CodeGeneration/Models/EnumerationItemModel.cs [hunk 6]
```diff
@@ -19,5 +19,8 @@ public class EnumerationItemModel
 
         /// <summary>Gets or sets the internal value (e.g. the underlying/system value).</summary>
         public string InternalValue { get; set; }
+        
+        /// <summary> A text description of what this entry means, eg CALL = a call option </summary>
```
Comment (RSuter) [inline src/NJsonSchema.CodeGeneration/Models/EnumerationItemModel.cs]: Start with "Gets the"

File: src/NJsonSchema.CodeGeneration.CSharp/Templates/Enum.liquid (hunk 5 above)
Comment (screig) [reply src/NJsonSchema.CodeGeneration.CSharp/Templates/Enum.liquid]: Not familiar with this liquid format, how does one do that?

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs (hunk 1 above)
Comment (screig) [reply src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: I'm not familiar with the JSON schema. I am processing a lot of JSON files and see these two 'annotations' all over the text.

Comment (screig) [reply src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: examples  https://github.com/ANNA-DSB/Product-Definitions/tree/master/PROD/JSON

File: src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs (hunk 2 above)
Comment (screig) [reply src/NJsonSchema.CodeGeneration.CSharp/Models/EnumTemplateModel.cs]: as above

**Comment by RicoSuter** [comment]
Very old PR, closing now. Use liquid template overrides for specific customizations.

**Pull request closed without merging**